client/my-sites/sidebar/sidebar.jsx

```diff
@@ -366,7 +366,7 @@ export class MySitesSidebar extends Component {
 			linkClass += ' is-paid-plan';
 		}
 
-		let planName = site.plan.product_name_short;
+		let planName = this.props.translate( site.plan.product_name_short );
```

Comment:
Oh nice catch here. We do have [translated plan names in the client](https://github.com/Automattic/wp-calypso/blob/master/client/lib/plans/constants.js#L107) if you feel up to the changes. It would look a bit like:

```
import { getPlan } from 'client/lib/plans';
//...
const planName = getPlan( plan.productSlug ).getTitle();
```